src/components/engagement/events/engagement-updated.event.ts

```diff
@@ -11,8 +11,8 @@ export class EngagementUpdatedEvent {
   constructor(
     public updated: UnsecuredDto<Engagement>,
     readonly previous: UnsecuredDto<Engagement>,
-    readonly input: UpdateLanguageEngagement | UpdateInternshipEngagement,
     readonly session: Session,
+    readonly input?: UpdateLanguageEngagement | UpdateInternshipEngagement,
```

Comment:
@CarsonF I'm not sure if changing the input param to optional has further impacts but based on high level scans I think this will be ok, but let me know if there is a better way to design the event publishing for engagement workflow.
@bryanjnelson @willdch